pkg/client/client_test.go

```diff
@@ -57,25 +60,47 @@ func TestApplySingle(t *testing.T) {
 		// given
 		namespacedName := types.NamespacedName{Namespace: "toolchain-host-operator", Name: "registration-service"}
 
-		t.Run("when using forceUpdate=true, it should update even when spec is same", func(t *testing.T) {
+		t.Run("when using forceUpdate=true, it should not update when specs are same", func(t *testing.T) {
 			// given
-			cli := NewFakeClient(t)
-			cl := applyCl.NewApplyClient(cli, s)
-			_, err := cl.CreateOrUpdateObject(defaultService.DeepCopyObject(), true, nil)
+			cl, _ := newClient(t, s)
+			obj := defaultService.DeepCopyObject()
+			_, err := cl.CreateOrUpdateObject(obj, true, nil)
 			require.NoError(t, err)
+			originalGeneration := obj.(*corev1.Service).GetGeneration()
 
-			// when
-			createdOrChanged, err := cl.CreateOrUpdateObject(modifiedService.DeepCopyObject(), true, nil)
+			// when updating with the same obj again
+			createdOrChanged, err := cl.CreateOrUpdateObject(obj, true, nil)
 
 			// then
 			require.NoError(t, err)
-			assert.True(t, createdOrChanged)
+			assert.False(t, createdOrChanged) // resource was not updated on the server, so returned value is `false`
+			updateGeneration := obj.(*corev1.Service).GetGeneration()
+			assert.Equal(t, updateGeneration, originalGeneration)
+		})
+
```

Comment:
```suggestion

		t.Run("when using forceUpdate=true, it will update even when specs are same and return true as labels have changed", func(t *testing.T) {
			// given
			cl, _ := newClient(t, s)
			service := defaultService.DeepCopy()
			_, err := cl.CreateOrUpdateObject(service, true, nil)
			require.NoError(t, err)
			originalGeneration := service.GetGeneration()
			service.SetLabels(map[string]string{"new": "label"})

			// when updating with changed labels
			createdOrChanged, err := cl.CreateOrUpdateObject(service, true, nil)

			// then
			require.NoError(t, err)
			assert.True(t, createdOrChanged) // resource was updated on the server, so returned value is `true`
			updateGeneration := service.GetGeneration()
			assert.Equal(t, originalGeneration+1, updateGeneration)
		})
```